src/JustSaying/Messaging/Middleware/Middleware`2.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Threading.Tasks;
+
+namespace JustSaying.Messaging.Middleware
+{
+    public abstract class MiddlewareBase<TContext, TOut>
```

Comment:
`MessageMiddleware` or `HandlerMiddleware`? `MiddlewareBase` just seems a bit too vague to me.

Reply:
This is a base middleware type that can be implemented to be used in any of these situations, so I think the vagarity is warranted

Reply:
But it doesn't say _what_ it's a middleware of. In a .NET context, it makes me think of HTTP pipeline middlewares, which this isn't.

Reply:
That's a fair point I think - would something like `MessagingMiddlewareBase` work better? 
here are the current implementations:
`DefaultSqsMiddleware`: Middleware for the receive pipeline, used by `IMessageReceiveBuffer`
`DelegateMiddleware`: Used to wrap new middleware
`ErrorHandlingMiddleware`: An example of using a middleware to handle errors in the receive pipeline
`PollyMiddleware`: An example of a policy that enables the use of a Polly `IAsyncPolicy`.